api_app/serializers.py

```diff
@@ -120,12 +116,57 @@ def get_permissions_map(self, created):
             "change_job": [*usr_groups],
         }
 
+
+class _JobCreateSerializerMixin(
+    ObjectPermissionsAssignmentMixin, serializers.ModelSerializer
+):
+    """
+    Base Serializer for Job create().
+    """
+
+    def get_permissions_map(self, created):
+        """
+        * 'view' permission is applied to all the groups the requesting user belongs to
+        if private is True.
+        * 'delete' permission is only given to the user who created the job
+        * 'change' permission is given to
+        """
+        rqst = self.context["request"]
+        current_user = rqst.user
+        usr_groups = current_user.groups.all()
+        if rqst.data.get("private", False):
+            view_grps = usr_groups
+        else:
+            view_grps = Group.objects.all()
+
+        return {
+            "view_job": [*view_grps],
+            "delete_job": [*usr_groups],
+            "change_job": [*usr_groups],
+        }
+
     def validate(self, data):
         # check and validate runtime_configuration
         runtime_conf = data.get("runtime_configuration", {})
         if runtime_conf and isinstance(runtime_conf, list):
             runtime_conf = json.loads(runtime_conf[0])
         data["runtime_configuration"] = runtime_conf
+
+        run_all_available_analyzers = data.get("run_all_available_analyzers", False)
+        analyzers_requested = data.get("analyzers_requested", [])
+        if run_all_available_analyzers:
+            if analyzers_requested:
+                raise serializers.ValidationError(
+                    {
+                        """
+                        analyzers_requested and run_all_available_analyzers
+                        cannot be used together
+                        """
+                    }
+                )
+
+            data["analyzers_requested"] = analyzers_controller.ALL_ANALYZERS
```

Comment:
Remove this line. And then in the `_analysis_request` function check the `if` condition again for this. (as was done before)